lib/elixir/lib/module/locals_tracker.ex

```diff
@@ -98,6 +98,17 @@ defmodule Module.LocalsTracker do
     {unreachable(reachable, reattached, private), collect_warnings(reachable, private)}
   end
 
+  def collect_undefined_locals({_set, bag}, all_defined) do
+    undefined =
+      Enum.reduce(all_defined, %{}, fn {pair, _, meta, _}, acc ->
+        undefined_from(bag, pair, meta, acc)
+      end)
+
+    for {pair, {:undefined_function, meta}} <- undefined do
+      {meta, {:undefined_function, pair}}
```

Comment:
Unfortunately I think this meta is not enough. We need to use the meta of the call site and not of the definition. We will probably have to change how things are stored in ETS.